clang/lib/AST/Type.cpp

```diff
@@ -2837,6 +2837,10 @@ bool QualType::hasNonTrivialToPrimitiveCopyCUnion(const RecordDecl *RD) {
   return RD->hasNonTrivialToPrimitiveCopyCUnion();
 }
 
+bool QualType::hasUninitializedExplicitInitFields(const RecordDecl *RD) {
```

Comment:
This is a bit of a weird one, why is this in `QualTYpe` but takes the thing anyway?  Should the implementation here be `getAsRecordDecl()->hasUnin...` ?

Reply:
Right, I would expect that.  I'm just a little shocked we have something like this that has nothing to do with `QualType` in `QualType`.  I DO note however that this doesn't seem to be called anywhere?  Is it actually NEEDED?